src/test/java/io/jenkins/plugins/sprp/AgentTest.java

```diff
@@ -0,0 +1,187 @@
+package io.jenkins.plugins.sprp;
+
+import hudson.Launcher;
+import io.jenkins.plugins.sprp.models.Agent;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+
+public class AgentTest {
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    private final Launcher launcher = (Launcher) jenkinsRule.createLocalLauncher();
+    private PipelineSnippetGenerator pipelineSnippetGenerator = new PipelineSnippetGenerator(launcher);
+
+    @Test
+    public void nodeGenerationTest() {
+        Agent agent = new Agent();
+        agent.setLabel("my-label");
+        agent.setCustomWorkspace("my-customWorkspace");
+
+        String agentSnippetActual =
+                pipelineSnippetGenerator.autoAddTabs((ArrayList<String>) pipelineSnippetGenerator.getAgent(agent));
+
+        String agentSnippetExpected =
+                "agent {\n" +
+                        "\tnode {\n" +
+                        "\t\tlabel 'my-label'\n" +
+                        "\t\tcustomWorkspace 'my-customWorkspace'\n" +
+                        "\t}\n" +
+                        "}\n";
+
+        assertEquals(agentSnippetExpected, agentSnippetActual);
+    }
+
+    @Test
+    public void dockerGenerationTest() {
+        Agent agent = new Agent();
+        agent.setDockerImage("my-docker-image");
+        agent.setArgs("-v temp:temp");
+
+        String agentSnippetActual =
+                pipelineSnippetGenerator.autoAddTabs((ArrayList<String>) pipelineSnippetGenerator.getAgent(agent));
+
+        String agentSnippetExpected =
+                "agent {\n" +
+                        "\tdocker {\n" +
+                        "\t\timage 'my-docker-image'\n" +
+                        "\t\targs '-v temp:temp'\n" +
+                        "\t\talwaysPull false\n" +
+                        "\t\treuseNode false\n" +
+                        "\t}\n" +
+                        "}\n";
+
+        assertEquals(agentSnippetExpected, agentSnippetActual);
+    }
+
+    @Test
+    public void dockerGenerationTest1(){
```

Comment:
The "test#" naming for tests isn't helpful. Instead of `dockerGenerationTest1` you could maybe do something like `dockerGenerationWithAlwaysPullTest`

Here's some more information on other naming conventions for tests in Java that might be useful to look over. 

https://stackoverflow.com/questions/155436/unit-test-naming-best-practices

https://dzone.com/articles/7-popular-unit-test-naming